src/Classes/ServiceAPI/MyRadio_Event.php

```diff
@@ -0,0 +1,243 @@
+
+    /**
+     * @var int|null
+     */
+    private $masterId;
+
+    public function __construct(array $data)
+    {
+        $this->eventid = (int)$data['eventid'];
+        $this->title = $data['title'];
+        $this->descriptionHtml = $data['description_html'];
+
+        $this->startTime = strtotime($data['start_time']);
+        $this->endTime = strtotime($data['end_time']);
+
+        $this->hostId = (int)$data['hostid'];
+
+        $this->rrule = $data['rrule'];
+
+        $this->masterId = is_null($data['master_id']) ? null : (int)$data['master_id'];
+    }
+
+    /**
+     * @return int
+     */
+    public function getID(): int
+    {
+        return $this->eventid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTitle(): string
+    {
+        return $this->title;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDescriptionHtml(): string
+    {
+        return $this->descriptionHtml;
+    }
+
+    /**
+     * @return int
+     */
+    public function getStartTime(): int
+    {
+        return $this->startTime;
+    }
+
+    /**
+     * @return int
+     */
+    public function getEndTime(): int
+    {
+        return $this->endTime;
+    }
+
+    /**
+     * @return int
+     */
+    public function getHostId(): int
+    {
+        return $this->hostId;
+    }
+
+    /**
+     * @return MyRadio_User
+     */
+    public function getHost(): MyRadio_User
+    {
+        return MyRadio_User::getInstance($this->hostId);
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getRrule(): ?string
+    {
+        return $this->rrule;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getMasterId(): ?int
+    {
+        return $this->masterId;
+    }
+
+    /**
+     * Get the next N events.
+     *
+     * @param $n int the number of events to get
+     * @return MyRadio_Event[] events
+     */
+    public static function getNext($n = 5)
+    {
+        $sql = 'SELECT eventid FROM public.events WHERE start_time >= NOW() ORDER BY start_time ASC LIMIT $1';
+        $rows = self::$db->fetchColumn($sql, [$n]);
+        return self::resultSetToObjArray($rows);
+    }
+
+    public static function getInRange($start, $end)
+    {
+        $sql = 'SELECT eventid FROM public.events WHERE start_time >= $1 AND end_time <= $2 ORDER BY start_time ASC';
+        $rows = self::$db->fetchColumn(
+            $sql,
+            [
+                CoreUtils::getTimestamp(strtotime($start)),
+                CoreUtils::getTimestamp(strtotime($end))
+            ]
+        );
+        return self::resultSetToObjArray($rows);
+    }
+
+    public static function create($data = [])
+    {
+        // Validate
+        $requiredFields = ['title', 'description_html', 'start_time', 'end_time'];
+        foreach ($requiredFields as $field) {
+            if (!(isset($data[$field]))) {
+                throw new MyRadioException("Missing $field", 400);
```

Comment:
rip my missing $field. $ in string

Reply:
it's a double-quote string, so this is fine :) it'll get replaced with e.g. `missing title`